__tests__/articles.test.js

```diff
@@ -55,47 +98,130 @@ describe("/api/articles/:article_id", () => {
+            expect(body.article.votes).toBe(initialVoteCount + 1);
+          });
+      });
+
+      it("Should decrement votes by 100", () => {
+        // First, fetch the initial vote count.
+        return request(app)
+          .get("/api/articles/1")
           .expect(200)
           .then(({ body }) => {
-            // Check if it's an array
-            expect(body.articles).toBeInstanceOf(Array);
+            const initialVotes = body.article.votes;
 
-            // Iterate through all articles and check their structure
-            body.articles.forEach((article) => {
-              expect(article).toMatchObject({
-                author: expect.any(String),
-                title: expect.any(String),
-                article_id: expect.any(Number),
-                topic: expect.any(String),
-                created_at: expect.any(String),
-                votes: expect.any(Number),
-                article_img_url: expect.any(String),
-                comment_count: expect.any(Number),
-              });
-              // Check for the absence of the body property for each article
-              expect(article).not.toHaveProperty("body");
+            // Then, send the PATCH request to decrement the votes.
+            return {
+              request: request(app)
+                .patch("/api/articles/1")
+                .send({ inc_votes: -100 })
+                .expect(200),
+              initialVotes: initialVotes,
+            };
+          })
+          .then(({ request, initialVotes }) => {
+            return request.then(({ body }) => {
+              expect(body.article.votes).toBe(initialVotes - 100);
             });
           });
       });
 
-      it("should return articles sorted by date in descending order", () => {
+      it("Responds with the updated article", () => {
+        // First, fetch the initial vote count.
         return request(app)
-          .get("/api/articles")
+          .get("/api/articles/1")
           .expect(200)
           .then(({ body }) => {
-            expect(body.articles).toBeSortedBy("created_at", {
-              descending: true,
+            const initialVotes = body.article.votes;
+
+            // Then, send the PATCH request to increment the votes by 5.
+            return {
+              request: request(app)
+                .patch("/api/articles/1")
+                .send({ inc_votes: 5 })
+                .expect(200),
+              initialVotes: initialVotes,
+            };
+          })
+          .then(({ request, initialVotes }) => {
+            return request.then(({ body }) => {
+              expect(body.article.votes).toBe(initialVotes + 5);
             });
           });
       });
+      it("should return an empty array of comments for an article without comments", (done) => {
+        request(app)
+          .get("/api/articles/2/comments")
+          .expect(200)
+          .end((err, res) => {
+            if (err) return done(err);
```

Comment:
If you've used promises throughout, its good to continue uing promises for conistentcy, instead of switching to using callbacks